CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptRecognizers.cs

```diff
@@ -0,0 +1,460 @@
+        public bool? ExcludeValue { get; set; }
+        /// <summary>
+        /// (Optional) If true, the choices action will NOT be recognized over.
+        /// </summary>
+        public bool? ExcludeAction { get; set; }
+        /// <summary>
+        /// (Optional) if true, then only some of the tokens in a value need to exist to be considered a match.The default value is "false".
+        /// </summary>
+        public bool? AllowPartialMatches { get; set; }
+        /// <summary>
+        /// (Optional) maximum tokens allowed between two matched tokens in the utterance.So with
+        /// a max distance of 2 the value "second last" would match the utternace "second from the last"
+        /// but it wouldn't match "Wait a second. That's not the last one is it?". 
+        /// The default value is "2".  
+        /// </summary>
+        public int? MaxTokenDistance { get; set; }
+    }
+    
+    public class LocalizedCache<T>
+    {
+        public string Id { get; set; }
+        public IDictionary<string, T> Locales { get; set; }
+
+        public LocalizedCache()
+        {
+            this.Locales = new Dictionary<string, T>();
+        }
+        public LocalizedCache(string key)
+            : this()
+        {
+            this.Id = key;
+        }
+    }
+
+    public class LocalizedCacheList<T> : List<LocalizedCache<T>>
+    {
+        public LocalizedCache<T> this[string key]
+        {
+            get { return this.SingleOrDefault(x => x.Id == key); }
+        }
+    }
+
+    public class ChronoDuration
+    {
+        public string Entity { get; internal set; }
+        public ChronoDurationResolution Resolution { get; set; }
+
+        public ChronoDuration()
+        {
+            this.Resolution = new ChronoDurationResolution();
+        }
+    }
+
+    public class ChronoDurationResolution
+    {
+        public string ResolutionType { get; set; }
+        public DateTime? Start { get; set; }
+        public DateTime? End { get; set; }
+    }
+
+    public class PromptRecognizers
+    {
+        private static LocalizedCacheList<Regex> expCache = new LocalizedCacheList<Regex>();
+        private static LocalizedCacheList<IDictionary<string, IEnumerable<string>>> choicesCache = new LocalizedCacheList<IDictionary<string, IEnumerable<string>>>();
+
+        public static IEnumerable<RecognizeEntity<string>> RecognizeLocalizedRegExp(IMessageActivity context, string expressionKey, ResourceManager resourceManager)
+        {
+            var entities = new List<RecognizeEntity<string>>();
+            var locale = context.Locale;
+            var utterance = context.Text ?? string.Empty;
+            var cache = expCache[expressionKey];
+            if (cache == null)
+            {
+                cache = new LocalizedCache<Regex>(expressionKey);
+                expCache.Add(cache);
+            }
+            if (string.IsNullOrEmpty(locale))
+            {
+                locale = string.Empty;
```

Comment:
It will default to English because later on it uses the ResourceManager of the Resource. In any case that code needs to be removed as it seems a leftover from a previous version (it doesn't make sense)